src/main/java/com/teamacronymcoders/epos/container/QuiverContainer.java

```diff
@@ -0,0 +1,118 @@
+package com.teamacronymcoders.epos.container;
+
+import com.hrznstudio.titanium.api.client.AssetTypes;
+import com.hrznstudio.titanium.block.tile.inventory.PosInvHandler;
+import com.hrznstudio.titanium.client.gui.asset.IAssetProvider;
+import com.teamacronymcoders.epos.feature.quiver.QuiverItem;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.inventory.container.Container;
+import net.minecraft.inventory.container.ContainerType;
+import net.minecraft.inventory.container.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.items.SlotItemHandler;
+import net.minecraftforge.registries.ObjectHolder;
+
+import java.awt.*;
+
+public class QuiverContainer extends Container {
+    @ObjectHolder("epos:quiver_container")
+    public static final ContainerType<QuiverContainer> TYPE = null;
+
+    private PosInvHandler handler;
+    private PlayerInventory player;
+    private boolean hasPlayerInventory;
+    private int totalSlots;
+
+    public QuiverContainer(int id, PlayerInventory playerInventory, PacketBuffer buffer) {
+        this(((QuiverItem) playerInventory.player.getHeldItemMainhand().getItem()).getHandler(playerInventory.player.getHeldItemMainhand()), playerInventory);
```

Comment:
Don't ever cast without checking.

Reply:
Also what if the quiver is in the off hand?

Reply:
I'm not sure how to do this since I can't run checks on it Sky.
Since it's both a "this" statement which has to be the first in the thing.
Also since this is a required default for building a container in 1.14+